### PR TITLE
Fix `Slider` overlay and value indicator interactive behavior on desktop.

### LINKED ISSUE #113367: `Slider` thumb overlay doesn't behave properly on desktop
### Description

When working on https://github.com/flutter/flutter/issues/111451, I found `Slider` doesn't behave properly on desktop platforms.
It behaves as it does on mobile, while `Slider` is slightly different in behavior on desktop. 

This is a comparison between Flutter and [MWC Slider demo](https://material-components.github.io/material-web/demos/slider/).

### 1.  Unfocused `Slider` shows thumb overlay when hovering outside the thumb.

When `Slider` is initially unfocused,  hovering outside the thumb area brings the thumb overlay.

#### Material Web

https://user-images.githubusercontent.com/48603081/195550171-be4d8b15-e27e-4711-a2bb-da74bf4cf81a.mov

#### Flutter

https://user-images.githubusercontent.com/48603081/195550204-13a14d2a-add2-4d57-a654-c17ca7f4ccbe.mov

#### Material Web

https://user-images.githubusercontent.com/48603081/195550906-5f1bd761-669d-44cc-8a2b-9f16af0e3206.mov

#### Flutter

https://user-images.githubusercontent.com/48603081/195550963-61084eb2-1a02-4532-a8a3-044ae9ed7d6b.mov

### PR DESCRIPTION
**Pull request by TahaTesser**
This is part of  [Update `Slider` to support Material 3](https://github.com/flutter/flutter/issues/111451).
fixes https://github.com/flutter/flutter/issues/113367

#### This PR fixes the following issues
(this is compared to https://material-components.github.io/material-web/demos/slider/)

#### 1. Unfocused `Slider` shows thumb overlay when hovering outside the thumb.

When `Slider` is not in focus, It should only show overlay when the pointer is hovering on the thumb.

#### Before

ezgif com-gif-maker (4)

#### After
ezgif com-gif-maker (2)

### 2.  The thumb overlay disappears when clicking on the `Slider`.

When you click to change the `Slider` value, it should get focus and the overlay should remain until `Slider` is unfocused

#### Before

ezgif com-gif-maker (5)

#### After

ezgif com-gif-maker (3)

### 2.  The value indicator disappears when clicking on the `Slider`.

When you click to change the `Slider` value, it should get focus and the value indicator should remain until `Slider` is unfocused

#### Before

ezgif com-gif-maker

#### After

ezgif com-gif-maker (1)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by HansMuller** [comment]
@guidezpl - could you verify that the new highlight behavior is M3-correct? It LGTM, just wanted to make sure.

**Comment by guidezpl** [comment]
This LGTM. The overlaid indicator should remain visible as well, is that the case?

**Comment by TahaTesser** [comment]
> This LGTM. The overlaid indicator should remain visible as well, is that the case?

That makes sense if Slider is discrete or `showValueIndicator: ShowValueIndicator.always`, the value indicator should remain visible.

**Comment by guidezpl** [comment]
It should also remain visible when focused, if it appears when the slider lobe is clicked.

File: packages/flutter/test/material/slider_test.dart [hunk 1]
```diff
@@ -3177,4 +3177,208 @@ void main() {
...
+      isNot(paints..path(color: const Color(0xff000000))..paragraph()),
+    );
+
+    final Offset sliderCenter = tester.getCenter(find.byType(Slider));
+    final Offset tapLocation = Offset(sliderCenter.dx + 50, sliderCenter.dy);
+
+    // Tap somewhere to bring overlay.
+    final TestGesture gesture = await tester.createGesture(kind: PointerDeviceKind.mouse);
+    await gesture.addPointer();
+    await gesture.down(tapLocation);
```
Comment (TahaTesser) [inline packages/flutter/test/material/slider_test.dart:3365]: I just fixed the value indicator that should remain when Slider is an interactive and added a test for this.

As this PR deals with interactions, request focus when interacting and keep overlay and value indicator unless unfocused.


> It should also remain visible when the slider is focused focused, if it appears when the slider lobe is clicked.

I filed an issue for this already https://github.com/flutter/flutter/issues/113538

`OverlayEntry` for value indicator is only created when you interact and focussed focused doesn't create `OverlayEntry`  for value indicator. I will fix this as part of  https://github.com/flutter/flutter/issues/113538.

Comment (TahaTesser) [reply packages/flutter/test/material/slider_test.dart:3365]: cc: @guidezpl 
Does this make sense? :)

Comment (TahaTesser) [reply packages/flutter/test/material/slider_test.dart:3365]: Also updated the [description](https://github.com/flutter/flutter/pull/113543#issue-1411786989) for the value indicator fix.

Comment (guidezpl) [reply packages/flutter/test/material/slider_test.dart:3365]: Gotcha, thanks makes sense

**Review by guidezpl — APPROVED** [review]

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**